python/deltalake/writer.py

```diff
@@ -202,6 +206,22 @@ def visitor(written_file: Any) -> None:
         )
 
 
+def __throw_error_if_delta_append_only_and_mode_not_append(
+    table: Optional[DeltaTable],
+    configuration: Optional[Mapping[str, Optional[str]]],
+    mode: str,
+) -> None:
```

Comment:
Could you please add a docstring to this method (you could also mention that it will throw an exception)?

Reply:
Please let me know if you want more in the docstring.